src/Domain/Beginner.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Domain;
+
+use PhpTuf\ComposerStager\Exception\DirectoryAlreadyExistsException;
+use PhpTuf\ComposerStager\Exception\DirectoryNotFoundException;
+use PhpTuf\ComposerStager\Infrastructure\Filesystem\Filesystem;
+use PhpTuf\ComposerStager\Infrastructure\Process\FileCopier;
+
+class Beginner implements BeginnerInterface
+{
+    /**
+     * @var \PhpTuf\ComposerStager\Infrastructure\Process\FileCopier
+     */
+    private $fileCopier;
+
+    /**
+     * @var \PhpTuf\ComposerStager\Infrastructure\Filesystem\Filesystem
+     */
+    private $filesystem;
+
+    public function __construct(FileCopier $fileCopier, Filesystem $filesystem)
+    {
+        $this->fileCopier = $fileCopier;
+        $this->filesystem = $filesystem;
+    }
+
+    public function begin(string $activeDir, string $stagingDir, ?callable $callback = null): void
+    {
+        if (!$this->activeDirectoryExists($activeDir)) {
+            throw new DirectoryNotFoundException($activeDir, 'The active directory does not exist at "%s"');
+        }
+
+        if ($this->stagingDirectoryExists($stagingDir)) {
+            throw new DirectoryAlreadyExistsException($stagingDir, 'The staging directory already exists at "%s"');
+        }
+
+        $exclusions = [
+            $stagingDir,
+            '.git',
+        ];
```

Comment:
These exclusions are hardcoded pending a decision on how clients will provide their own.

Reply:
Will you add a `@todo` maybe, so we remember to change that later?